### PR TITLE
Refactor peep update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
ride sub state 0

### CODE REVIEW

File: src/openrct2/peep/Peep.cpp [hunk 1]
```diff
@@ -2695,8 +2822,8 @@ static void peep_update_ride_sub_state_0(rct_peep * peep)
         }
     }
 
-    uint8 car_array_size = 0xFF;
-    uint8 car_array[255];
+    uint8 carArraySize = 0xFF;
+    uint8 carArray[255];
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: Why does this declare a var `carArraySize`, but does not use it for the size of `carArray`?

File: src/openrct2/peep/Peep.cpp [hunk 2]
```diff
@@ -2655,7 +2655,7 @@ static void peep_go_to_ride_entrance(rct_peep * peep, Ride * ride)
 
     peep_decrement_num_riders(peep);
     peep->state     = PEEP_STATE_ENTERING_RIDE;
-    peep->sub_state = 1;
+    peep->sub_state = PEEP_RIDE_SS_IN_ENTRANCE;
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: SS?

File: src/openrct2/ride/Vehicle.h [hunk 3]
```diff
@@ -251,7 +251,7 @@ enum {
     VEHICLE_ENTRY_FLAG_VEHICLE_ANIMATION = 1 << 23,             // Set on animated vehicles like the Multi-dimension coaster trains, Miniature Railway locomotives and Helicycles.
     VEHICLE_ENTRY_FLAG_RIDER_ANIMATION = 1 << 24,               // Set when the animation updates rider sprite positions
     VEHICLE_ENTRY_FLAG_25 = 1 << 25,
-    VEHICLE_ENTRY_FLAG_26 = 1 << 26,
+    VEHICLE_ENTRY_FLAG_XY_LOADING_POSITIONS = 1 << 26,
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.h]: The meaning of this isn't immediately clear. It could use a comment.

File: src/openrct2/peep/Peep.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/peep/Peep.cpp]: The array will be of variable size at this point we don't know. It is guaranteed to be at most 255 though

File: src/openrct2/peep/Peep.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/openrct2/peep/Peep.cpp]: Sub state. It was getting too ridiculously long the name

File: src/openrct2/ride/Vehicle.h (hunk 3 above)
Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.h]: Yeah sorry i was in a rush to save my work when i wrote this. I'm still working out how best to do the peep loading positions. There are two different structures deepening on the state of this flag. Can't decide how best to expose that in the vehicle structure.

File: src/openrct2/peep/Peep.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/peep/Peep.cpp]: But carArray is declared _immediately_ after carArraySize...

Comment (duncanspumpkin) [reply src/openrct2/peep/Peep.cpp]: Well either way it should be a vector.

**Comment by duncanspumpkin** [comment]
There is still more work to do on this but it should be fine for merging.

File: src/openrct2/peep/Peep.cpp [hunk 4]
```diff
@@ -2673,14 +2673,141 @@ static void peep_go_to_ride_entrance(rct_peep * peep, Ride * ride)
...
  */
-static void peep_update_ride_sub_state_0(rct_peep * peep)
+static void peep_update_ride_ss_at_entrance(rct_peep * peep)
 {
     Ride * ride = get_ride(peep->current_ride);
 
+    // The peep will keep advancing in the entranceway
+    // whilst in this state. When it has reached the very
+    // front of the queue destination tolerance is set to 
+    // zero to indicate it is final decision time (try_leave will pass).
```
Comment (Broxzier) [inline src/openrct2/peep/Peep.cpp]: This state is also used for guests that exit though the entrance back to the front of the queue, which may be nice to clarify here.

File: src/openrct2/peep/Peep.cpp [hunk 5]
```diff
@@ -2378,7 +2380,7 @@ static void peep_update_falling(rct_peep * peep)
     peep->next_y = peep->y & 0xFFE0;
     peep->next_z = saved_map->base_height;
 
-    sint32 edx = saved_map->properties.surface.slope & TILE_ELEMENT_SLOPE_W_CORNER_DN;
+    sint32 edx = saved_map->properties.path.type & 0x7;
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: Is there really no define or function for this? I'm trying to get rid of direct access to tile element fields.

Comment (Broxzier) [reply src/openrct2/peep/Peep.cpp]: There are two parts combined here: `FOOTPATH_PROPERTIES_FLAG_IS_SLOPED | FOOTPATH_PROPERTIES_SLOPE_DIRECTION_MASK`

File: src/openrct2/peep/Peep.cpp [hunk 6]
```diff
@@ -4419,7 +4436,7 @@ static void peep_update_ride_sub_state_20(rct_peep * peep)
     {
         if (peep->nausea <= 35)
         {
-            peep->sub_state++;
+            peep->sub_state = 21;
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: raw number.

File: src/openrct2/peep/Peep.cpp [hunk 7]
```diff
@@ -4504,80 +4521,69 @@ static void peep_update_ride(rct_peep * peep)
...
-        peep_update_ride_sub_state_8(peep);
+    case PEEP_RIDE_APPROACH_EXIT:
+        peep_update_ride_approach_exit(peep);
         break;
-    case 9:
-        peep_update_ride_sub_state_9(peep);
+    case PEEP_RIDE_IN_EXIT:
+        peep_update_ride_in_exit(peep);
         break;
     case 10:
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: raw numbers

File: src/openrct2/peep/Peep.cpp [hunk 8]
```diff
@@ -5800,7 +5806,6 @@ static void peep_update_1(rct_peep * peep)
     peep->destination_x         = peep->x;
     peep->destination_y         = peep->y;
     peep->destination_tolerance = 10;
-    peep->var_76                = 0;
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: Why this change?

Comment (duncanspumpkin) [reply src/openrct2/peep/Peep.cpp]: It's not used at any point. I did quite a search on this before removing it

File: src/openrct2/peep/Peep.cpp (hunk 5 above)
Comment (duncanspumpkin) [reply src/openrct2/peep/Peep.cpp]: Yeah this was a ninja edit so that I don't forget that this was wrong. I want to go through and finally demystifi next var 27

File: src/openrct2/peep/Peep.cpp (hunk 7 above)
Comment (duncanspumpkin) [reply src/openrct2/peep/Peep.cpp]: They don't exist. I think I'll just get rid of them all together
